lib/views/create-bug.js

```diff
@@ -80,38 +79,33 @@ function drawAttachments() {
 // API needs anyway, may want to avoid encoding until they are sent
 function inputChanged(e) {
   var files = e.target.files;
+  var queue = [];
   for (var i = 0; i < files.length; i++) {
-    var file = files.item(i);
-    var reader = new FileReader();
-    reader.onload = (function(theFile) {
-      return function(e) {
-        bug.attachments.push({
-          name: theFile.name,
-          type: theFile.type,
-          data: e.target.result.split(',')[1]
-        });
-        bugChanged();
-        drawAttachments();
-      }
-    })(file);
-    reader.readAsDataURL(file);
+    queue.push(pushAttachment(files.item(i)));
   }
+  Promise.all(queue).then(bugChanged);
 }
 
 
-function addAttachments(blobs, names) {
-  blobs.forEach(function(blob, i) {
-    var reader = new window.FileReader();
-    reader.readAsDataURL(blob);
-    reader.onloadend = function() {
+function addActivityAttachments(blobs, names) {
+  Promise.all(blobs.map(function(blob, i) {
+    return pushAttachment(blob, names[i]);
+  })).then(bugChanged);
+}
+
+
+function pushAttachment(file, name) {
+  return new Promise(function(resolve) {
+    var reader = new FileReader();
```

Comment:
```FileReader``` has an ```onerror``` callback. We should hook this up to a ```reject()``` call. This is especially important since we're handling arbitrary user data here.